### PR TITLE
[gperftools] bump version from 2.7 to 2.9.1

### PR DESCRIPTION
**Pull request by Crghilardi**
This PR bumps `gperftools` from 2.7 up to 2.9.1. I also tried to update it a bit with wider platforms and `julia_compat` statement.

### CODE REVIEW

File: G/gperftools/build_tarballs.jl [hunk 1]
```diff
@@ -3,11 +3,11 @@
 using BinaryBuilder, Pkg
 
 name = "gperftools"
-version = v"2.7.0"
+version = v"2.9.1"
 
 # Collection of sources required to complete build
 sources = [
-    ArchiveSource("https://github.com/gperftools/gperftools/releases/download/gperftools-2.7/gperftools-2.7.tar.gz", "1ee8c8699a0eff6b6a203e59b43330536b22bbcbe6448f54c7091e5efb0763c9")
+    ArchiveSource("https://github.com/gperftools/gperftools/releases/download/gperftools-2.9.1/gperftools-2.9.1.tar.gz","ea566e528605befb830671e359118c2da718f721c27225cbbc93858c7520fee3")
```
Comment (jeremiahpslewis) [inline G/gperftools/build_tarballs.jl:10]:
```suggestion
    ArchiveSource("https://github.com/gperftools/gperftools/releases/download/gperftools-$(version)/gperftools-$(version).tar.gz","ea566e528605befb830671e359118c2da718f721c27225cbbc93858c7520fee3")
```

File: G/gperftools/build_tarballs.jl [hunk 2]
```diff
@@ -24,28 +24,27 @@ elif [[ "${target}" == *-apple-* ]]; then
...
-    Platform("i686", "linux"; libc="glibc"),
-    Platform("x86_64", "linux"; libc="glibc"),
-    Platform("aarch64", "linux"; libc="glibc"),
-    Platform("armv7l", "linux"; libc="glibc"),
-    Platform("powerpc64le", "linux"; libc="glibc"),
-    Platform("x86_64", "macos"),
-    Platform("x86_64", "freebsd")
-]
-platforms = expand_cxxstring_abis(platforms)
+platforms = expand_cxxstring_abis(supported_platforms(; experimental = true))
```
Comment (jeremiahpslewis) [inline G/gperftools/build_tarballs.jl:46]:
```suggestion
platforms = expand_cxxstring_abis(supported_platforms(; exclude=Sys.iswindows))
```


`experimental=true` is now the default so I don't think this needs to be specified any longer. :)

File: G/gperftools/build_tarballs.jl [hunk 3]
```diff
@@ -24,28 +24,27 @@ elif [[ "${target}" == *-apple-* ]]; then
...
+filter!(!Sys.iswindows, platforms)
```
Comment (jeremiahpslewis) [inline G/gperftools/build_tarballs.jl:47]:
```suggestion
```

File: G/gperftools/build_tarballs.jl [hunk 4]
```diff
@@ -24,28 +24,26 @@ elif [[ "${target}" == *-apple-* ]]; then
...
-    Platform("i686", "linux"; libc="glibc"),
-    Platform("x86_64", "linux"; libc="glibc"),
-    Platform("aarch64", "linux"; libc="glibc"),
-    Platform("armv7l", "linux"; libc="glibc"),
-    Platform("powerpc64le", "linux"; libc="glibc"),
-    Platform("x86_64", "macos"),
-    Platform("x86_64", "freebsd")
-]
-platforms = expand_cxxstring_abis(platforms)
+platforms = expand_cxxstring_abis(supported_platforms(; exclude=Sys.iswindows))
```
Comment (giordano) [inline G/gperftools/build_tarballs.jl:47]: I don't think we can do much for aarch64-linux-musl, judging by the error messages in the log I think it's an upstream bug.  We never built for that platform before, I'd just filter it out
```suggestion
filter!(p -> !(Sys.islinux(p) && arch(p) = 64 && libc(p) == "musl"), platforms)
```

Comment (Crghilardi) [reply G/gperftools/build_tarballs.jl:49]: That sounds good, I thought that might be the case. The same thing happened to me on windows, I took one or two looks at it and then just filtered it out

**Pull request merged**